components/hab/src/command/origin/create.rs

```diff
@@ -0,0 +1,33 @@
+use crate::{api_client::{self,
+                         Client},
+            common::ui::{Status,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            PRODUCT,
+            VERSION};
+use hyper::status::StatusCode;
+
+pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Creating, format!("origin {}.", origin))?;
+
+    match api_client.create_origin(origin, token) {
+        Ok(_) => {
+            ui.status(Status::Created, format!("origin {}.", origin))
+              .map_err(Error::from)
+        }
+        Err(api_client::Error::APIError(StatusCode::Conflict, msg)) => {
```

Comment:
I don't think we should fail on a conflict return - instead we can just indicate that the origin already exists, similar to how origin key upload handles conflicts.  This will also make it much easier to use in the on-prem scripts.